src/benchmarks/gc/GC.Infrastructure/GC.Infrastructure.NotebookTests/Utils.cs

```diff
@@ -0,0 +1,134 @@
+                    process.Start();
+                    process.WaitForExit();
+                }
+
+                catch (System.ComponentModel.Win32Exception)
+                {
+                    return false;
+                }
+
+                return process.ExitCode == 0;
+            }
+        }
+
+        public static void InstallDotnetRepl()
+        {
+            using (Process process = new Process())
+            {
+                process.StartInfo.FileName = "dotnet";
+                process.StartInfo.Arguments = "tool install --local dotnet-repl";
+                process.StartInfo.UseShellExecute = false;
+                process.Start();
+                process.WaitForExit();
+            }
+        }
+
+        public static Process SetupNotebookProcessRun(string notebookPath, string? outputNotebookPath = null)
+        {
+            Process process = new Process();
+            process.StartInfo.FileName = "dotnet-repl";
+            process.StartInfo.UseShellExecute = false;
+            process.StartInfo.Arguments = $"--run {notebookPath} --exit-after-run";
+
+            // Optionally include the output notebook path in case we care about the output.
+            if (!string.IsNullOrEmpty(outputNotebookPath))
+            {
+                process.StartInfo.Arguments += $" --output-path {outputNotebookPath}";
+            }
+
+            // The working directory should be the directory that the notebook is in.
+            process.StartInfo.WorkingDirectory = Path.GetDirectoryName(notebookPath);
+            return process;
+        }
+
+        public static void RunNotebookThatsExpectedToPass(string notebookPath) 
+        {
+            string tempPathForOutputNotebook = Path.GetTempFileName();
+            using (Process dotnetReplProcess = SetupNotebookProcessRun(notebookPath, tempPathForOutputNotebook))
+            {
+                dotnetReplProcess.Start();
+                dotnetReplProcess.WaitForExit(TIMEOUT);
+
+                // If notebook execution fails, parse the notebook and find the error.
+                if (dotnetReplProcess.ExitCode != 0)
+                {
+                    ParseNotebookAndFindErrors(tempPathForOutputNotebook, notebookPath);
+                }
+
+                else
+                {
+                    dotnetReplProcess.HasExited.Should().BeTrue();
+                }
+            }
+
+            // Cleanup any temp notebooks.
+            File.Delete(tempPathForOutputNotebook);
+        }
+
+        public static void RunNotebookThatsExpectedToFail(string notebookPath)
+        {
+            string tempPathForOutputNotebook = Path.GetTempFileName();
+            using (Process dotnetReplProcess = SetupNotebookProcessRun(notebookPath, tempPathForOutputNotebook))
+            {
+                dotnetReplProcess.Start();
+                dotnetReplProcess.WaitForExit(TIMEOUT);
+                dotnetReplProcess.ExitCode.Should().NotBe(0);
+            }
```

Comment:
This path doesn't seem to delete the temp notebook.  I wonder if it's worth expanding that helper function to also execute/wait for the process and then delete the file.  I guess it would need a processing lambda to do stuff before deleting the file.

If not, then I think you want the `Delete` in a `finally` so that it always executes.

Reply:
But first, should the output notebooks be deleted?  When run under automation, typically files are left behind as artifacts for later inspection, either for all tests or for failing tests.  They also might be better stored out of the source tree.

Reply:
Will address saving and retrieving notebook outputs in the future.